src/Mod/Draft/DraftSnap.py

```diff
@@ -1268,127 +1302,118 @@ def cancel():
+        """
+        builds the Snap toolbar"""
         mw = FreeCADGui.getMainWindow()
         self.toolbar = QtGui.QToolBar(mw)
         mw.addToolBar(QtCore.Qt.TopToolBarArea, self.toolbar)
         self.toolbar.setObjectName("Draft Snap")
         self.toolbar.setWindowTitle(QtCore.QCoreApplication.translate("Workbench", "Draft Snap"))
-        self.toolbarButtons = []
-        # grid button
-        self.gridbutton = QtGui.QAction(mw)
-        self.gridbutton.setIcon(QtGui.QIcon.fromTheme("Draft_Grid", QtGui.QIcon(":/icons/Draft_Grid.svg")))
-        self.gridbutton.setText(QtCore.QCoreApplication.translate("Draft_ToggleGrid","Grid"))
-        self.gridbutton.setToolTip(QtCore.QCoreApplication.translate("Draft_ToggleGrid","Toggles the Draft grid On/Off"))
-        self.gridbutton.setObjectName("GridButton")
-        self.gridbutton.setWhatsThis("Draft_ToggleGrid")
-        QtCore.QObject.connect(self.gridbutton,QtCore.SIGNAL("triggered()"),self.toggleGrid)
-        self.toolbar.addAction(self.gridbutton)
-        # master button
-        self.masterbutton = QtGui.QAction(mw)
-        self.masterbutton.setIcon(QtGui.QIcon.fromTheme("Snap_Lock", QtGui.QIcon(":/icons/Snap_Lock.svg")))
-        self.masterbutton.setText(QtCore.QCoreApplication.translate("Draft_Snap_Lock","Lock"))
-        self.masterbutton.setToolTip(QtCore.QCoreApplication.translate("Draft_Snap_Lock","Toggle On/Off"))
-        self.masterbutton.setObjectName("SnapButtonMain")
-        self.masterbutton.setWhatsThis("Draft_ToggleSnap")
-        self.masterbutton.setCheckable(True)
-        self.masterbutton.setChecked(True)
-        QtCore.QObject.connect(self.masterbutton,QtCore.SIGNAL("toggled(bool)"),self.toggle)
-        self.toolbar.addAction(self.masterbutton)
-        for c,i in self.cursors.items():
-            if i:
-                b = QtGui.QAction(mw)
-                b.setIcon(QtGui.QIcon.fromTheme(i.replace(':/icons/', '').replace('.svg', ''), QtGui.QIcon(i)))
-                if c == "passive":
-                    b.setText(QtCore.QCoreApplication.translate("Draft_Snap_Near","Nearest"))
-                    b.setToolTip(QtCore.QCoreApplication.translate("Draft_Snap_Near","Nearest"))
-                else:
-                    b.setText(QtCore.QCoreApplication.translate("Draft_Snap_"+c.capitalize(),c.capitalize()))
-                    b.setToolTip(QtCore.QCoreApplication.translate("Draft_Snap_"+c.capitalize(),c.capitalize()))
-                b.setObjectName("SnapButton"+c)
-                b.setWhatsThis("Draft_"+c.capitalize())
-                b.setCheckable(True)
-                b.setChecked(True)
-                self.toolbar.addAction(b)
-                self.toolbarButtons.append(b)
-                QtCore.QObject.connect(b,QtCore.SIGNAL("toggled(bool)"),self.saveSnapModes)
-        # adding non-snap button
-        for n in ["Dimensions","WorkingPlane"]:
+
+        from draftutils.init_tools import get_draft_snap_commands
```

Comment:
Imports should be at the beginning of the files. I am moving some imports around in all the pending pull requests.

Some imports I still haven't moved, but hopefully these will come eventually.